lib/spack/spack/spec.py

```diff
@@ -188,8 +188,10 @@
                  '^': dependency_color,
                  '#': hash_color}
 
-"""Regex used for splitting by spec field separators."""
-_separators = '[%s]' % ''.join(color_formats.keys())
+"""Regex used for splitting by spec field separators.
+These need to be escaped to avoid metacharacters in color_formats.keys().
+"""
```

Comment:
Your docstring is on the wrong side of `_separators` for the docs. Looks like this is true for many parts of Spack. See [here](http://www.sphinx-doc.org/en/stable/ext/autodoc.html#directive-autoattribute):

> For module data members and class attributes, documentation can either be put into a comment with special formatting (using a `#:` to start the comment instead of just `#`), or in a docstring _after_ the definition. Comments need to be either on a line of their own _before_ the definition, or immediately after the assignment _on the same line_. The latter form is restricted to one line only.

As it is, this docstring is used for `color_formats`.